### PR TITLE
add check_stmt_for_find so that we can implement `find` rules separately 

### PR DESCRIPTION
**Pull request by nnhoang**

### CODE REVIEW

File: src/semant.ml [hunk 1]
```diff
@@ -98,8 +94,39 @@ let check (contexts, finds) =
...
   let check_find findBlk =
-    check_have_var findBlk.A.fcontext findBlk.A.ftarget;
-    List.iter check_stmt findBlk.A.fbody
+    let symbolmap =
+      let ctx_name = findBlk.A.fcontext in
+      try StringMap.find ctx_name varmap
+      with Not_found -> fail ("unrecognized context, " ^ quot ctx_name)
+    in
+
+  (* Verify a statement or throw an exception *)
```
Comment (jzacsh) [inline src/semant.ml]: "[...] statement for a particular find expression [...]"

Comment (rxie25) [inline src/semant.ml]: it looks good to me :laughing:

File: src/semant.ml [hunk 2]
```diff
@@ -98,8 +94,39 @@ let check (contexts, finds) =
...
+  let rec check_stmt_for_find = function
+      | A.Block sl ->
+          (* effectively unravel statements out of their block *)
+          let rec check_block = function
```
Comment (jzacsh) [inline src/semant.ml]: (same as earlier chat we had - can you unravel this `let foo = .... in foo arg;` pattern?)

File: src/semant.ml [hunk 3]
```diff
@@ -43,11 +43,7 @@ let check (contexts, finds) =
 
   let varmap = List.fold_left create_varmap StringMap.empty contexts in
 
-  let check_have_var ctx_name var =
-    let symbolmap =
-      try StringMap.find ctx_name varmap
-      with Not_found -> fail ("unrecognized context, " ^ quot ctx_name)
-    in
+  let check_have_var var symbolmap =
```
Comment (jzacsh) [inline src/semant.ml]: maybe these would be better to read/write if they were `get_context` or `get_var`. so perhaps:
1. rename this `check_have_var` to `get_var`
2. rename `symbolmap` below to `get_context`

Sound silly?

**Comment by jzacsh** [comment]
looks great, all readability-comments! commit when you want

**Pull request merged**